### DISCUSSION TITLE
Sharing/embedding + passwords can be a little confusing for new users testing integration

### DISCUSSION THREAD

**Discussion by chrisbennett-Bene** [plausible/analytics#1491] [Anything Else]
Love your work, started testing yesterday, and will be subscribing long before trial expires.

One thing I noticed - or perhaps didn't notice - as a first-time user, was the information bubbles.
Big fan of the clean, minimal design aesthetic, but in my case it caused a little confusion when it came to testing.

Specifically, because my brain had apparently tuned out the (quite prominent) info links, when it came to testing generated embeds I went around in circles for a while. 

Basically, it was a bit too easy for me to 'accidentally' fill the password field on the **New Shared Link Page** out of habit (where no extra info is provided). My password manager, ignoring autocomplete off, offered to generate a password for me.

I foolishly didn't realise it would be useless to generate an embed based on that shared link.
After longer than I'd care to admit, I **finally** grasped my mistake and slapped my forehead.

I think it may have helped to warn a little more explicitly on the **New Shared Link Page** itself. 
Either or both of the following possible alterations might have helped me understand what was going "wrong" a bit sooner.

**Add placeholder to password field:**
`<input autocomplete="off" class="shadow-sm focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 rounded-md" id="shared_link_password" name="shared_link[password]" type="password" placeholder="Leave blank to share without password">`

Entirely possible my Password Manager, and possibly others, would ignore that placeholder anyway and just focus on the type="password", so not sure how useful that would have been by itself. Probably wouldn't hurt wither way.

**Slight tweak to notes under password field:**
```
<p class="mt-2 text-sm text-gray-500">
<span class="text-gray-400">Password protection is optional</span>. 
Please save it securely. Once the link is created, we cannot reveal the password.
</p>
<p class="mt-2 text-sm text-gray-400">
<strong>Note:</strong> Embedded dashboards will not work with a password. 
</p>
```

With a little editing along those lines, you could add the extra warning, with a modest increase in height of around 5px or so.

![idea](https://user-images.githubusercontent.com/47375905/143323295-041fe853-c073-48b4-8bca-2eba14621212.png)

**Comment by ukutaht** [comment]
Thanks @chrisbennett-Bene!

Makes sense, it's not easy to find this information. Personally I like the addition of the `Note: ` line but I'm not a fan of the placeholder text. I'll add it

**Reply by chrisbennett-Bene** [reply]
Awesome @ukutaht

**Comment by RobertJoonas** [comment]
Another report on this confusion via email:

> When creating a shared link, my browser auto-fills with my Plausible
> account email and password. This is potentially
> confusing and it would be a good idea if you were to change the field
> names and autofill settings to avoid this.

Perhaps we can also disable browser auto-filling these fields?

**Comment by chrisbennett-Bene** [comment]
I can't say for sure what happened with other people, but for me it was my password manager rather than browser.

I think as soon as it saw  `id="shared_link_password" name="shared_link[password]" type="password"` it was always going to ignore the autocomplete="off".

A possible workaround might be to load the password field as type="text" (possibly might need to change id and name as well) then swap input type (and id/name if necessary) on focus / change.

**Reply by chrisbennett-Bene** [reply]
Should have finished my first coffee of the day before pressing send ;)

Another, possibly cleaner and more understandable option:

You could load the password field and accompanying explanation/warning of what password protection will not allow, when a user toggles a nicely styled 'Password protect this link' checkbox.

**Reply by ukutaht** [reply]
Yes I like that approach, really nice and clean
